client/components/domains/transfer-domain-step/index.jsx

```diff
@@ -132,16 +134,23 @@ class TransferDomainStep extends React.Component {
 	};
 
 	addTransfer() {
-		const { translate } = this.props;
+		const { cart, domainsWithPlansOnly, selectedSite, translate } = this.props;
 		const { searchQuery, submittingAvailability, submittingWhois } = this.state;
 		const submitting = submittingAvailability || submittingWhois;
 		const productSlug = getDomainProductSlug( searchQuery );
-		const domainProductPrice = getDomainPrice(
+
+		let domainProductPrice = getDomainPrice(
 			productSlug,
 			this.props.productsList,
 			this.props.currencyCode
 		);
 
+		if ( isNextDomainFree( cart ) || isDomainBundledWithPlan( cart, searchQuery ) ) {
+			domainProductPrice = translate( 'Free with your plan' );
+		} else if ( domainsWithPlansOnly && selectedSite && ! isPlan( selectedSite.plan ) ) {
```

Comment:
This fixed the error, but now the price is wrong:
<img width="684" alt="screen shot 2018-05-16 at 12 13 59" src="https://user-images.githubusercontent.com/3392497/40111315-c6347b62-5902-11e8-972c-62e329a343f5.png">

As for the new registration, it should say "Included in paid plans", not the price for just the registration. This is in signup - so DWPO is true, of course. Tested also with an taken (but transferrable) domain, same result.